maths/carmichael_number.py

```diff
@@ -31,15 +34,38 @@ def power(x: int, y: int, mod: int) -> int:
 
 
 def is_carmichael_number(n: int) -> bool:
+    """
+
+    Examples:
+    >>> is_carmichael_number(562)
+    False
+
+    >>> is_carmichael_number(561)
+    True
+
+    >>> is_carmichael_number(5.1)
+    Traceback (most recent call last):
+         ...
+    ValueError: Number 5.1 must instead be integer
+    """
+
+    if not isinstance(n, int):
+        msg = f"Number {n} must instead be integer"
+        raise ValueError(msg)
+
     b = 2
```

Comment:
All of this could be simplified:
```py
return all(
    power(b, n - 1, n) == 1
    for b in range(2, n) if greatest_common_divisor(b, n) == 1
)
```